frontend/src/Editor/Components/Table/Filter.jsx

```diff
@@ -93,6 +107,46 @@ export function Filter(props) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [JSON.stringify(filters)]);
 
+  React.useEffect(() => {
+    setExposedVariable('setFilters', async function (_filters) {
+      if (!isArray(_filters)) return;
+      const filterArr = [];
+      _filters.forEach((_filter) => {
+        const { column = '', value = '', condition = '' } = _filter;
+        const columnId = props.columns.find((col) => col.name === column)?.value;
+        const isCorrectCondition = FILTER_OPTIONS.some((option) => option.value === condition);
+        if (columnId && isCorrectCondition) {
+          const filterObj = {
+            id: columnId,
+            value: {
+              column,
+              condition,
+              value,
+            },
+          };
+          filterArr.push(filterObj);
+        }
+      });
+      if (filterArr.length) {
+        setAllFilters(filterArr.filter((filter) => filter.id !== ''));
+        mergeToFilterDetails({
+          filters: filterArr,
+        });
+      }
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(props.columns)]);
+
+  React.useEffect(() => {
+    setExposedVariable('clearFilters', async function (_filters) {
+      setAllFilters([]);
+      mergeToFilterDetails({
+        filters: [],
+      });
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(props.columns)]);
```

Comment:
Can we combine both in a single `useEffect` as the dependency is the same? And can we some comments about what's the CSA - [ref](https://github.com/ToolJet/ToolJet/pull/8674/files#diff-6303cb67e509da2bc5279261ca27dc50be29c03ab7c2ba1dcaf6b57a42cb8122R887)?